app/controllers/graders_controller.rb

```diff
@@ -123,10 +123,10 @@ def global_actions
           end
         end
         if found_empty_submission
-          randomly_assign_graders(filtered_grouping_ids, grader_ids)
+          randomly_assign_graders(filtered_grouping_ids, grader_ids, params[:weightings])
```

Comment:
I'd like to see a bit of error-checking here. Convert all of the weightings to floats here, and if one of the conversion fails, flash an error message (you'll need to create a new one) and respond with a `head :bad_request` as above. That way you can pass in a list of floats to `randomly_assign_graders` and don't need to worry about conversions in that function.

Reply:
Oh and one more error check: all of the weightings should be `>= 0`, and their sum should be `> 0`. (We can't simply rely on client-side checking, we should check this on the server as well.)

Reply:
And finally, please add some new controller test cases for these checks :)

Reply:
According to https://www.rubydoc.info/stdlib/core/String:to_f converting to a float cannot fail. Do you still want me to add a test just in case? Or alternatively do you want me to manually test for invalid strings?

Reply:
@WeissSa aha, yes great point! Use https://ruby-doc.org/core-3.1.2/Kernel.html#method-i-Float instead (which can raise an exception if the argument doesn't represent a float)

Reply:
Oh okay! Sounds good